test/testpaint/Compat.cpp

```diff
@@ -174,152 +174,204 @@ rct_tile_element* map_get_first_element_at(int x, int y)
+    return (colour & TRACK_ELEMENT_DOOR_A_MASK) >> 2;
 }
 
-bool is_csg_loaded()
+uint8_t TrackElement::GetDoorBState() const
 {
-    return false;
+    return (colour & TRACK_ELEMENT_DOOR_B_MASK) >> 5;
+}
+
+uint8_t TrackElement::GetRideIndex() const
+{
+    return rideIndex;
 }
 
-uint8_t track_element_get_colour_scheme(const rct_tile_element* tileElement)
+void TrackElement::SetRideIndex(uint8_t newRideIndex)
 {
-    return tileElement->properties.track.colour & 0x3;
+    rideIndex = newRideIndex;
 }
 
-uint16_t track_element_get_maze_entry(const rct_tile_element* tileElement)
+uint8_t TrackElement::GetColourScheme() const
 {
-    return tileElement->properties.track.maze_entry;
+    return colour & 0x3;
 }
 
-uint8_t track_element_get_ride_index(const rct_tile_element* tileElement)
+void TrackElement::SetColourScheme(uint8_t newColourScheme)
 {
-    return tileElement->properties.track.ride_index;
+    colour &= ~0x3;
+    colour |= (newColourScheme & 0x3);
 }
 
-void track_element_set_ride_index(rct_tile_element* tileElement, uint8_t rideIndex)
+bool TrackElement::HasCableLift() const
 {
-    tileElement->properties.track.ride_index = rideIndex;
+    return colour & TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
 }
 
-uint8_t track_element_get_type(const rct_tile_element* tileElement)
+void TrackElement::SetHasCableLift(bool on)
 {
-    return tileElement->properties.track.type;
+    colour &= ~TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
+    if (on)
+        colour |= TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
 }
 
-void track_element_set_type(rct_tile_element* tileElement, uint8_t type)
+bool TrackElement::IsInverted() const
 {
-    tileElement->properties.track.type = type;
+    return colour & TRACK_ELEMENT_COLOUR_FLAG_INVERTED;
 }
 
-void track_element_set_cable_lift(rct_tile_element* trackElement)
+void TrackElement::SetInverted(bool inverted)
 {
-    trackElement->properties.track.colour |= TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
+    if (inverted)
+    {
+        colour |= TRACK_ELEMENT_COLOUR_FLAG_INVERTED;
+    }
+    else
+    {
+        colour &= ~TRACK_ELEMENT_COLOUR_FLAG_INVERTED;
+    }
+}
+
+uint8_t TrackElement::GetBrakeBoosterSpeed() const
+{
+    return (sequence >> 4) << 1;
+}
+
+void TrackElement::SetBrakeBoosterSpeed(uint8_t speed)
+{
+    sequence &= ~0b11110000;
+    sequence |= ((speed >> 1) << 4);
+}
+
+uint8_t TrackElement::HasGreenLight() const
+{
+    return (sequence & MAP_ELEM_TRACK_SEQUENCE_GREEN_LIGHT) != 0;
+}
+
+void TrackElement::SetHasGreenLight(uint8_t greenLight)
+{
+    sequence &= ~MAP_ELEM_TRACK_SEQUENCE_GREEN_LIGHT;
+    if (greenLight)
+    {
+        sequence |= MAP_ELEM_TRACK_SEQUENCE_GREEN_LIGHT;
+    }
+}
+
+bool TrackElement::HasChain() const
+{
+    return type & TRACK_ELEMENT_TYPE_FLAG_CHAIN_LIFT;
 }
 
-void track_element_clear_cable_lift(rct_tile_element* trackElement)
+void TrackElement::SetHasChain(bool on)
 {
-    trackElement->properties.track.colour &= ~TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
+    if (on)
```

Comment:
Might be able to make use of xor (^) here.

Reply:
I prefer the current method for consistency.